### ISSUE TITLE
Multi-line translations in `.stringsdict` file not being processed

### ISSUE THREAD

**Issue by Jeehut** [SwiftGen/SwiftGen#915]
When there's an entry like this in a `.stringsdict` file (copied the DuckDuckGo iOS project [here](https://github.com/duckduckgo/iOS/blob/develop/DuckDuckGo/de.lproj/Localizable.stringsdict#L5-L28)):

```xml
        <key>dax.onboarding.browsing.multiple.trackers</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%1#@count@</string>
		<key>count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>d</string>
			<key>one</key>
			<string>*%2$@, %3$@* und * 1 anderes Tracker-Netzwerk* haben versucht, dich hier zu tracken.

Ich habe sie blockiert!

 ☝️Du kannst in der URL-Leiste überprüfen, wer versucht, dich zu tracken, wenn du eine neue Website besuchst.</string>
			<key>other</key>
			<string>*%2$@, %3$@* und *%1$d andere Tracker-Netzwerke* haben versucht, dich hier zu tracken.

Ich habe sie blockiert!

 ☝️Du kannst in der URL-Leiste überprüfen, wer versucht, dich zu tracken, wenn du eine neue Website besuchst.</string>
		</dict>
	</dict>
```

Then the `StringsDict` type results in this object (note the `variables` having 0 elements, although there should be one):

```swift
- key : "dax.onboarding.browsing.multiple.trackers"
      ▿ value : StringsDict
        ▿ pluralEntry : PluralEntry
          - formatKey : "%1#@count@"
          - variables : 0 elements
```

It seems like the decoding somehow doesn't recognize the entire variable named `count` here due to these multi-line values. Thus, SwiftGen doesn't produce a correct `enum` here. Any ideas for a fix? Is this an issue with `PropertyListDecoder` or with the `Decodable` type structure? Not sure ...

**Comment by Jeehut** [comment]
After playing a bit around with the translations, I figured the issue isn't even related to the multi-line string, but to the fact that the key starts with `%1#@` instead of `%#@`, which seems not to be supported by SwiftGen (although it works).

**Comment by Jeehut** [comment]
Okay, looks like SwiftGen does have support for the positional variants, but only in the style `%1$@` with the `$` in there, which seems to be missing in the DuckDuckGo `stringsdict` file linked above. It sill works without the `$`, so Xcode seems to accept both styles. Maybe we should then also accept it without the `$`? Let me create a PR to fix it.

Unfortunately, I can't find clear documentation on this, if you can point me to the official docs on positional ones: 🙌

**Comment by djbe** [comment]
After some digging found the specs again.
- This page documents all the "string format specifiers" supported by Foundation: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Strings/Articles/formatSpecifiers.html#//apple_ref/doc/uid/TP40004265-SW1
- There's a short mention about "positionals", which links through to https://pubs.opengroup.org/onlinepubs/009695399/functions/printf.html
- That clearly only mentions the format `%n$` for positionals

The slight complication here is the stringsdict variable syntax, which uses `%#@blabla@`. But the syntax remains the same: for positionals, add `n$` directly after the `%`, so `%1$#@blabla@`. Quite a bunch of symbols, but that should be it. Our own code docs mention this: https://github.com/SwiftGen/SwiftGen/blob/stable/Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift#L138-L139

So the specs as far as I can find any documentation for it, say the syntax you mentioned "should" be invalid. The IEEE docs only talk about `%n$` 🤷

**Comment by Jeehut** [comment]
@djbe Thanks for looking things up. I guess, then there's no need to change it and we could close both this thread and my PR #916 (which I didn't do properly anyways due to time restrictions). Unless you want to make the SwiftGen parser here accept the same invalid syntax that also Xcode seems to accept. Your choice. :)

**Comment by djbe** [comment]
Well your PR demonstrates it IS possible (I assume), it would need a bunch of extra test cases to be sure.

I'm just wondering *where* this is coming from. From a quick check, top 6 search results related to stringsdict & positional use the correct syntax. Funnily enough the 7th is our [own strings documentation](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/Parsers/strings.md).

**Comment by djbe** [comment]
I'm going to close this issue for now, as I've yet to find official docs to support this feature.

If anyone can find new docs that prove the contrary, please reopen this issue.

**Comment by Jeehut** [comment]
I'm no longer using SwiftGen for Strings files (as I wrote [my own app](https://www.fline.dev/remafox-beta1/) instead, which is why I found this issue in the first place during testing), but for the sake of SwiftGen users I think having the less strict Regex supporting the same styles that Xcode seems to support does not have any downsides. If Xcode ever removes support for this style, developers need to change it anyways. But of course the decision is not mine and I respect if you have different reasons not to merge.

Thanks for considering and updating in any case!
